homeassistant/components/sensor/travisci.py

```diff
@@ -0,0 +1,173 @@
+"""
+This component provides HA sensor support for Travis CI framework.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.travisci/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_SCAN_INTERVAL, CONF_MONITORED_CONDITIONS,
+    STATE_UNKNOWN)
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['TravisPy==0.3.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ATTRIBUTION = "Information provided by https://travis-ci.org/"
+CONF_BRANCH_NAME = 'default_branch'
+CONF_GITHUB_TOKEN = 'github_token'
+CONF_REPOSITORY_NAME = 'repository_names'
+
+DEFAULT_BRANCH_NAME = 'master'
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+# sensor_type [ description, unit, icon ]
+SENSOR_TYPES = {
+    'last_build_id': ['Last Build ID', '', 'mdi:account-card-details'],
+    'last_build_duration': ['Last Build Duration', 'sec', 'mdi:timelapse'],
+    'last_build_finished_at': ['Last Build Finished At', '', 'mdi:timetable'],
+    'last_build_started_at': ['Last Build Started At', '', 'mdi:timetable'],
+    'last_build_state': ['Last Build State', '', 'mdi:github-circle'],
+    'state': ['State', '', 'mdi:github-circle'],
+
+}
+
+NOTIFICATION_ID = 'travisci'
+NOTIFICATION_TITLE = 'Travis CI Sensor Setup'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_GITHUB_TOKEN): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Required(CONF_BRANCH_NAME, default=DEFAULT_BRANCH_NAME): cv.string,
+    vol.Optional(CONF_REPOSITORY_NAME, default=[]):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+        cv.time_period,
+})
```

Comment:
I think you should use the `CONF_API_KEY` from const.py and simplify names with `CONF_BRANCH -> branch` and `CONF_REPOSITORY -> repository`